cmd/validate/policy.go

```diff
@@ -0,0 +1,94 @@
+// Copyright The Enterprise Contract Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package validate
+
+import (
+	"context"
+	"fmt"
+
+	hd "github.com/MakeNowJust/heredoc"
+	"github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+
+	"github.com/enterprise-contract/ec-cli/internal/policy"
+	validate_utils "github.com/enterprise-contract/ec-cli/internal/validate"
+)
+
+type policyValidationFunc func(context.Context, string) error
+
+func ValidatePolicyCmd(validate policyValidationFunc) *cobra.Command {
+	var data = struct {
+		policyConfiguration string
+		output              []string
+		strict              bool
+	}{
+		strict: true,
+	}
+	cmd := &cobra.Command{
+		Use:   "policy",
+		Short: "Validate the provided EnterpriseContractPolicy spec",
+		Long: hd.Doc(`
+			Validate the provided EnterpriseContractPolicy spec against the EnterpriseContractPolicy spec schema used in this version of the ec CLI
+		`),
+		Example: hd.Doc(`
+			Validate a local policy configuration file:
+			ec validate policy --policy-configuration policy.yaml
+
+			Validate a policy configuration file from a github repository:
+			ec validate policy --policy-configuration github.com/org/repo/policy.yaml
+`),
+		PreRunE: func(cmd *cobra.Command, args []string) (allErrors error) {
+			if len(policy.ECPSchema) == 0 {
+				return fmt.Errorf("ECP schema is not defined")
+			}
+
+			ctx := cmd.Context()
+
+			policyConfiguration, err := validate_utils.GetPolicyConfig(ctx, data.policyConfiguration)
+			if err != nil {
+				allErrors = multierror.Append(allErrors, err)
+				return
+			}
+			data.policyConfiguration = policyConfiguration
+
+			return
+
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Policy conforms to the schema.
+			ctx := cmd.Context()
+			err := validate(ctx, data.policyConfiguration)
+			if err != nil {
+				return fmt.Errorf("policy configuration does not conform to the EnterpriseContractPolicy spec")
+			}
+			fmt.Println("Policy configuration conforms to the EnterpriseContractPolicy spec")
```

Comment:
We want to use the writer configured for the command here

```suggestion
			fmt.Fprintln(cmd.OutOrStdout(), "Policy configuration conforms to the EnterpriseContractPolicy spec")
```

Reply:
Ack. Updated